test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go

```diff
@@ -7,41 +7,165 @@ package integrationslogs
+	Logs       []LogsConfig `yaml:"logs"`
+}
 
-//go:embed fixtures/tenLogs.yaml
-var writeTenLogsConfig string
+type Instance struct {
+	LogMessage      string `yaml:"log_message"`
+	UniqueMessage   bool   `yaml:"unique_message"`
+	LogSize         int    `yaml:"log_size"`
+	LogCount        int    `yaml:"log_count"`
+	IntegrationTags string `yaml:"integration_tags"`
+}
+
+type LogsConfig struct {
+	Type    string `yaml:"type"`
+	Source  string `yaml:"source"`
+	Service string `yaml:"service"`
+}
 
 // TestLinuxFakeIntakeSuite
 func TestIntegrationsLogsSuite(t *testing.T) {
 	suiteParams := []e2e.SuiteOption{
 		e2e.WithProvisioner(awshost.Provisioner(awshost.WithAgentOptions(
 			agentparams.WithLogs(),
-			// set the integration log file max size to 1MB
-			agentparams.WithAgentConfig("logs_config.integrations_logs_files_max_size: 1"),
-			agentparams.WithFile("/etc/datadog-agent/checks.d/writeTenLogs.py", writeTenLogsCheck, true),
-			agentparams.WithFile("/etc/datadog-agent/conf.d/writeTenLogs.yaml", writeTenLogsConfig, true))))}
+			agentparams.WithAgentConfig("logs_config.integrations_logs_files_max_size: 1"))))}
+
+	suiteParams = append(suiteParams, e2e.WithDevMode())
 
 	e2e.Run(t, &IntegrationsLogsSuite{}, suiteParams...)
 }
 
 // TestWriteTenLogsCheck ensures a check that logs are written to the file ten
 // logs at a time
 func (v *IntegrationsLogsSuite) TestWriteTenLogsCheck() {
-	utils.CheckLogsExpected(v.T(), v.Env().FakeIntake, "ten_logs_service", "Custom log message", []string{"env:dev", "bar:foo"})
+	tags := []string{"foo:bar", "env:dev"}
+	yamlData, err := generateYaml("Custom log message", false, 1, 10, tags, "logs_from_integrations_source", "logs_from_integrations_service")
+	assert.NoError(v.T(), err)
+
+	v.UpdateEnv(awshost.Provisioner(awshost.WithAgentOptions(
+		agentparams.WithLogs(),
+		agentparams.WithFile("/etc/datadog-agent/conf.d/writeTenLogs.yaml", string(yamlData), true),
+		agentparams.WithFile("/etc/datadog-agent/checks.d/writeTenLogs.py", customIntegration, true))))
+
+	logs, err := utils.FetchAndFilterLogs(v.Env().FakeIntake, "logs_from_integrations_service", "Custom log message")
+	assert.Nil(v.T(), err)
+	assert.GreaterOrEqual(v.T(), len(logs), 10)
+}
+
+// TestIntegrationLogFileRotation ensures logs are captured after a integration
+// log file is rotated
+func (v *IntegrationsLogsSuite) TestIntegrationLogFileRotation() {
+	// Since it's not yet possible to write to the integration log file by calling
+	// the agent check command, we can test if the file rotation works using the following method:
+
+	// 1. Set the max log file size to 1 MB individual log size to 256 kB.
+	// 2. Send five (or more) logs to the agent, causing the log file to rotate.
+	// 3. Check the logs to ensure that each is unique, ensuring the rotation worked correctly.
+
+	tags := []string{"test:rotate"}
+	yamlData, err := generateYaml("a", true, 1024*230, 1, tags, "rotation_source", "rotation_service")
```

Comment:
should `1024*230` be `1024*256`? (or maybe `256 * 1000`? the actual default max log size) or is this just an arbitrarily high number? (either way this should be consistent with the comment above)

Reply:
I first experimented with 256 KB and that caused the logs to rotate before hitting the 1 MB threshold (it would rotate every 3 logs instead of every 4). I'm still not sure why that is considering the integration launcher should only take into account the size of actual log and not the other extraneous information (tags, source, service, etc.). This could be the source of a future investigation. I then chose this as an arbitrarily high number in order to ensure rotation every four logs, though that isn't as important now since the test is structured slightly differently, I'll update the comment with the size.

Reply:
We write the logs to the file in JSON with the tags as it's own field right?
So the the size mismatch could be from the JSON tags. 

If you can confirm this, let's update the comment to mention that.

Reply:
Ah, you're right. After debugging the integration launcher, I see that the log length includes tags, parentheses, and basically everything else included in the JSON formatted log, not just the message. Updating the comment now to reflect this. Thanks.